### ISSUE TITLE
RSC-005 Incorrect error for paragraph inside figure

### ISSUE THREAD

**Issue by kmuncie** [w3c/epubcheck#1299]
EPUBCheck is reporting`RSC-005 Error while parsing file: element "p" not allowed here; expected the element end-tag` when it likely should not. In my testing this markup renders normally in the major EPUB readers despite the EPUBCheck error.

The following markup is generating this error:

```html
<figure>
    <img src="../img/idpflogo_web_125.jpg" alt="Alt text" />
    <figcaption>
       <p>This is the figcaption</p>
    </figcaption>
    <p>This is a separate paragraph tag</p>
</figure>
```
This markup should be valid as `<figure>` can accept flow content. https://html.spec.whatwg.org/multipage/grouping-content.html#the-figure-element

I am attaching a test EPUB which has the markup shown above and is reporting the error at ./EPUB/xhtml/epub30-titlepage.xhtml (line 14, col 16)

[figure-test.epub.zip](https://github.com/w3c/epubcheck/files/7945417/figure-test.epub.zip)

**Comment by dauwhe** [comment]
> This markup should be valid as `<figure>` can accept flow content. https://html.spec.whatwg.org/multipage/grouping-content.html#the-figure-element

The content model for `figure` according to the HTML spec is: 

> Either: one figcaption element followed by flow content.
> Or: flow content followed by one figcaption element.
> Or: flow content.

My interpretation is that this appears to be flow content followed by one figcaption element followed by flow content. The HTML validator says this is invalid.

**Comment by rdeltour** [comment]
I concur with @dauwhe.

I'm closing this issue as `invalid` as this is apparently not a bug with EPUBCheck. But feel free to keep on commenting here, or reopen if you do think there's a bug somewhere.

**Comment by kmuncie** [comment]
Thanks for the response @dauwhe. I guess I am confounded by the spec then as it only appears to limit to one `figcaption` but any flow content (`img` and `p`) would be valid. 

Additionally in the examples listed below the spec is this one which shows a `<p>` tag inside the `<figure>`:

```html
<figure>
 <p>'Twas brillig, and the slithy toves<br>
 Did gyre and gimble in the wabe;<br>
 All mimsy were the borogoves,<br>
 And the mome raths outgrabe.</p>
 <figcaption><cite>Jabberwocky</cite> (first verse). Lewis Carroll, 1832-98</figcaption>
</figure>
```

**Comment by dauwhe** [comment]
Yeah, it's... weird. I did a test with `img` followed by `p` followed by `figcaption` and it was fine. Maybe I'll go ask on Twitter if anyone knows the rationale.
